cogito/ghcommitsink.go

```diff
@@ -90,3 +91,11 @@ func ghMakeContext(request PutRequest) string {
 	}
 	return context
 }
+
+func apiEndpoint(h string) string {
```

Comment:
maybe the better place would be to move this functions to github pkg?

Reply:
Yes it should go in the github package.
Note also that the name `apiEndpoint` is wrong, this is not returning an endpoint, it is returning the root or prefix of all the endpoints. Considering it being in package github, it could be something like

`github.ApiPrefix`
or
`github.ApiRoot`

I tend to think that `ApiRoot` is better ...

Reply:
Ah. I am now confused, I was expecting this function to append also the `/api/v3` (or whatever it was) prefix.

Reply:
What is the intent of this function, compared to where we add the `/api/v3` prefix? Can it be merged into one?

Note also that
```
	if strings.Contains(hostname, "127.0.0.1") {
		return fmt.Sprintf("http://%s", hostname)
	}
```
can fail in spectacular ways...

Reply:
This will arrive in the next PR... The next PR should really introduce the support for GH Enterprise (see the PR description).

I separated them to fix first the existing issues, and later concentrate on GH Enterprise feature only.

Reply:
Ah, yes.. Can you also comment a bit more on:

```
if strings.Contains(hostname, "127.0.0.1") {
		return fmt.Sprintf("http://%s", hostname)
	}
```
> can fail in spectacular ways...

As I see it; `127.0.0.1` is only configured from tests if this is a HTTP testserver. What am I missing here? 

In that ghinstance code from gh cli they did:
```
// Localhost is the domain name of a local GitHub instance.
const localhost = "github.localhost"

if strings.EqualFold(hostname, localhost) {
    return fmt.Sprintf("http://api.%s/", hostname)
}
```
since our localhost is always of the format: `127.0.0.1:PORT` I couldn't use the `==`

Reply:
Example: `127.0.0.1.hello.com` will match the conditional, and also `foo.127.0.0.1.hello.com`
In this case it is harmless, but in general one needs to be careful when manipulating user input.

Reply:
Fixed.